src/main/java/in/projecteka/gateway/link/discovery/DiscoveryController.java

```diff
@@ -0,0 +1,50 @@
+package in.projecteka.gateway.link.discovery;
+
+import in.projecteka.gateway.clients.DiscoveryServiceClient;
+import in.projecteka.gateway.registry.ServiceType;
+import in.projecteka.gateway.registry.YamlRegistryMapping;
+import in.projecteka.gateway.registry.YamlRegistrySource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import reactor.core.publisher.Mono;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+public class DiscoveryController {
+    public static final String X_HIP_ID = "X-HIP-ID";
+
+    @Autowired
+    YamlRegistrySource yamlRegistrySource;
```

Comment:
Can we name the class and variable better - like just bridgeRegistry? if you want to make it specific for HIU and HIP, then we can make it hiuRegistry or hipRegistry?

Reply:
Done.